src/helper/class-ee-site.php

```diff
@@ -506,16 +506,23 @@ public function restart( $args, $assoc_args, $whitelisted_containers = [] ) {
 
 		$this->site_data = get_site_info( $args );
 
-		chdir( $this->site_data['site_fs_path'] );
-
 		if ( $all || $no_service_specified ) {
 			$containers = $whitelisted_containers;
 		} else {
 			$containers = array_keys( $assoc_args );
 		}
 
-		foreach ( $containers as $container ) {
-			\EE\Site\Utils\run_compose_command( 'restart', $container );
+		if ( in_array( 'nginx', $containers ) ) {
+			$nginx_test_command = "sh -c 'nginx -t'";
+			if ( ! \EE_DOCKER::docker_compose_exec( $this->site_data['site_fs_path'], 'nginx', $nginx_test_command ) ) {
+				throw new \Exception( 'There was some error in docker-compose exec.' );
```

Comment:
A better error message on failure would be: `Nginx configuration test failed. Please check the site's nginx config.`
`There was some error in docker-compose exec` is a very generic message and does not convey anything helpful to the end user using EasyEngine.